bindings/experimental/distrdf/python/DistRDF/Ranges.py

```diff
@@ -51,330 +50,199 @@ def _n_even_chunks(iterable, n_chunks):
-        partition of the distributed dataset.
-
-        Args:
-            treename (str): Name of the tree.
-
-            filelist (list): List of ROOT files.
-
-            friend_info (FriendInfo): Information about friend trees.
-
-        Returns:
-            list[collections.namedtuple]: List containinig the ranges in which
-            the dataset has been split for distributed execution. Each ``Range``
-            contains a starting entry, an ending entry, the list of files
-            that are traversed to get all the entries and information about
-            friend trees::
-
-                [
-                    Range(start=0,
-                        end=42287856,
-                        filelist=['Run2012B_TauPlusX.root',
-                                  'Run2012C_TauPlusX.root'],
-                        friend_info=None),
-                    Range(start=6640348,
-                        end=51303171,
-                        filelist=['Run2012C_TauPlusX.root'],
-                        friend_info=None)
-                ]
-
-        """
-
-        # Retrieve a list of clusters for all files of the tree
-        clustersinfiles = self.get_clusters(treename, filelist)
-        numclusters = len(clustersinfiles)
-
-        # Restrict `npartitions` if it's greater than clusters of the dataset
-        if self.npartitions > numclusters:
-            msg = ("Number of partitions is greater than number of clusters "
-                   "in the dataset. Using {} partition(s)".format(numclusters))
-            warnings.warn(msg, UserWarning, stacklevel=2)
-            self.npartitions = numclusters
-
-        logger.debug("%s clusters will be split along %s partitions.",
-                     numclusters, self.npartitions)
+def get_clusters(treename, filelist):
+    """
+    Extract a list of cluster boundaries for the given tree and files
+    Args:
+        treename (str): Name of the TTree split into one or more files.
+        filelist (list): List of one or more ROOT files.
+    Returns:
+        list: List of tuples defining the cluster boundaries. Each tuple
+        contains four elements: first entry of a cluster, last entry of
+        cluster (exclusive), offset of the cluster and file where the
+        cluster belongs to::
+            [
+                (0, 100, 0, ("filename_1.root", 0)),
+                (100, 200, 0, ("filename_1.root", 0)),
+                ...,
+                (10000, 10100, 10000, ("filename_2.root", 1)),
+                (10100, 10200, 10000, ("filename_2.root", 1)),
+                ...,
+                (n, n+100, n, ("filename_n.root", n)),
+                (n+100, n+200, n, ("filename_n.root", n)),
+                ...
+            ]
+    """
 
-        """
-        This list comprehension builds ``Range`` tuples with the following
-        elements:
+    clusters = []
+    cluster = collections.namedtuple(
```

Comment:
Should we have these types created once (and with a nice name) instead of on each invocation of `get_clusters`?

Reply:
I agree, also for `EntriesRange` and `TreeRange`, see https://github.com/root-project/root/pull/8391#discussion_r650083804